covid19-notebooks/X-ray/peregrine.py

```diff
@@ -0,0 +1,93 @@
+import json
+import requests
+import pandas as pd
+import shutil
+
+url = "https://chicagoland.pandemicresponsecommons.org/"
+
+
+def get_token():
+    with open("/home/jovyan/pd/credentials.json", "r") as f:
+        creds = json.load(f)
+    token_url = url + "user/credentials/api/access_token"
+    token = requests.post(token_url, json=creds).json()["access_token"]
+    return token
+
+
+def query_api(query_txt, variables=None):
+    """ Request results for a specific query """
+    if variables == None:
+        query = {"query": query_txt}
+    else:
+        query = {"query": query_txt, "variables": variables}
+
+    request_url = url + "api/v0/submission/graphql"
+    output = requests.post(
```

Comment:
It would be better to use the guppy endpoint if that has the data you need.
Peregrine can be slow, and will get slower as more data goes into the graph.
Pauline can help with structuring the guppy query ...